rskj-core/src/main/java/co/rsk/RskContext.java

```diff
@@ -847,12 +848,34 @@ public synchronized RskModule getRskModule() {
                     getBlockchain(),
                     getBlockStore(),
                     getReceiptStore(),
-                    getWeb3InformationRetriever());
+                    getWeb3InformationRetriever(),
+                    getFlusher());
         }
 
         return rskModule;
     }
 
+    public synchronized BlockChainFlusher getBlockChainFlusher() {
+        checkIfNotClosed();
+
+        if (this.blockChainFlusher == null) {
+            this.blockChainFlusher = new BlockChainFlusher(
+                    getRskSystemProperties().flushNumberOfBlocks(),
+                    getCompositeEthereumListener(),
+                    getTrieStore(),
+                    getBlockStore(),
+                    getReceiptStore(),
+                    getBlocksBloomStore(),
+                    getStateRootsStore());
+        }
+
+        return blockChainFlusher;
+    }
+
+    public Flusher getFlusher() {
```

Comment:
Minor thing, but do we need this method if we only have the `BlockchainFlusher`? It's a bit confusing, it makes me thing we might have several and this method would "select" one.